browser/de_amp/de_amp_browsertest.cc

```diff
@@ -0,0 +1,278 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "base/bind.h"
+#include "base/path_service.h"
+#include "brave/browser/de_amp/de_amp_service_factory.h"
+#include "brave/common/brave_paths.h"
+#include "brave/components/de_amp/browser/de_amp_service.h"
+#include "brave/components/de_amp/common/features.h"
+#include "chrome/browser/profiles/keep_alive/profile_keep_alive_types.h"
+#include "chrome/browser/profiles/keep_alive/scoped_profile_keep_alive.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_commands.h"
+#include "chrome/browser/ui/browser_list.h"
+#include "components/keep_alive_registry/keep_alive_types.h"
+#include "components/keep_alive_registry/scoped_keep_alive.h"
+#include "components/prefs/testing_pref_service.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/content_mock_cert_verifier.h"
+#include "content/public/test/test_navigation_observer.h"
+#include "net/dns/mock_host_resolver.h"
+#include "net/test/embedded_test_server/embedded_test_server.h"
+#include "third_party/re2/src/re2/re2.h"
+
+#if defined(OS_ANDROID)
+#include "chrome/test/base/android/android_browser_test.h"
+#else
+#include "chrome/test/base/in_process_browser_test.h"
+#include "chrome/test/base/ui_test_utils.h"
+#endif
+
+const char kTestHost[] = "a.test.com";
+const char kTestAmpPage[] = "/test.html";
+const char kTestSimpleNonAmpPage[] = "/simple.html";
+const char kTestCanonicalPage[] = "/simple_canonical.html";
+const char kTestBody[] =
+    "<html amp>\n"
+    "<head>\n"
+    "<link rel='canonical' "
+    "href='https://%s:%s%s'>\n"
+    "</head></html>";
+class DeAmpBrowserTest : public PlatformBrowserTest {
+ public:
+  DeAmpBrowserTest() {
+    feature_list_.InitAndEnableFeature(de_amp::features::kBraveDeAMP);
+  }
+
+  void SetUpOnMainThread() override {
+    mock_cert_verifier_.mock_cert_verifier()->set_default_result(net::OK);
```

Comment:
Do we really need https? http is not enough?
for http we already has `embedded_test_server()` here.

Reply:
I don't know if we have a convention around this in tests, but I figured in a mostly-HTTPS world it makes sense to test on HTTPS URLs.